overviews/repl/embedding.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: overview-large
+title: Programmatic Use of Scala REPL
+
+disqus: true
+
+partof: repl
+num: 2
+outof: 2
+---
+
+The REPL can be embedded and invoked programmatically.
+
+It supports the `javax.script` API, or it can be used
+as either a code interpreter or an interactive command line.
+
+    import scala.tools.nsc.Settings
+    import scala.tools.nsc.interpreter._
+    import javax.script._
+
+    /** A simple example showing programmatic usage of the REPL. */
+    object Main extends App {
+
+      // the REPL has some support for javax.script
+      val scripter = new ScriptEngineManager().getEngineByName("scala")
+      scripter.eval("""println("hello, world")""")
+
+      // compiler settings
+      val settings = new Settings
+      settings.processArgumentString("-deprecation -feature -Xfatal-warnings -Xlint")
+
+      // the interpreter is used by the javax.script engine
+      val intp = new IMain(settings)
+      def interpret(code: String): Unit = {
+        import Results._
+        val res = intp.interpret(code) match {
+          case Success => "OK!"
+          case _       => "Sorry, try again."
+        }
+        println(res)
+      }
+      interpret("""println("hello, world")""")
+      interpret("""println(""")
+      interpret("""val who = "world" ; println("hello, $who")""")
```

Comment:
Yes, that was to show the warning output. It should probably explain expected output.